pkg/controller/csr_check.go

```diff
@@ -361,23 +360,21 @@ func isApproved(csr *certificatesv1beta1.CertificateSigningRequest) bool {
 	return false
 }
 
-func recentlyPendingCSRs(indexer cache.Indexer) int {
+func RecentlyPendingCSRs(csrs []certificatesv1beta1.CertificateSigningRequest) int {
 	// assumes we are scheduled on the master meaning our clock is the same
 	now := time.Now()
 	start := now.Add(-maxPendingDelta)
 	end := now.Add(maxMachineClockSkew)
 
 	var pending int
 
-	for _, item := range indexer.List() {
-		csr := item.(*certificatesv1beta1.CertificateSigningRequest)
-
+	for _, csr := range csrs {
 		// ignore "old" CSRs
 		if !inTimeSpan(start, end, csr.CreationTimestamp.Time) {
 			continue
 		}
 
-		if !isApproved(csr) {
+		if !isApproved(&csr) {
```

Comment:
Taking the address of a range variable is bad, it can lead to the wrong values coming out https://developmentality.wordpress.com/2014/02/25/go-gotcha-0-why-taking-the-address-of-an-iterated-variable-is-wrong/

If you need to take the address, copy into a temporary variable first